frontend/src/metabase/admin/settings/components/EmbeddingSettings/EmbeddingSdkLegaleseModal/EmbeddingSdkLegaleseModal.tsx

```diff
@@ -1,38 +1,26 @@
 import { useState } from "react";
 import { t } from "ttag";
 
-import {
-  Button,
-  Group,
-  List,
-  Modal,
-  type ModalProps,
-  Text,
-  Title,
-} from "metabase/ui";
+import { useAdminSetting } from "metabase/api/utils";
+import { Button, Group, List, Modal, type ModalProps, Text } from "metabase/ui";
 
-import type { AdminSettingComponentProps } from "../types";
-
-export const EmbeddingSdkLegaleseModal = ({
-  opened,
-  onClose,
-  updateSetting,
-}: AdminSettingComponentProps & ModalProps) => {
+export const EmbeddingSdkLegaleseModal = ({ opened, onClose }: ModalProps) => {
   const [loading, setLoading] = useState(false);
+  const { updateSettings } = useAdminSetting("show-sdk-embed-terms");
```

Comment:
Looks like we just need the `useAdminSetting` for the `updateSettings` props and we're not really using its values of `show-sdk-embed-terms`. I wonder if there is a more specific `useUpdateSettings` hook we can use instead.